policy-controller/k8s/status/src/index.rs

```diff
@@ -233,3 +252,33 @@ pub(crate) fn make_patch(
     });
     k8s::Patch::Merge(value)
 }
+
+pub async fn create_lease_manager(client: k8s::Client) -> Result<kubert::lease::LeaseManager> {
```

Comment:
Just a thought: why not roll this into the actual deployment as a k8s resource? It can simplify the logic a bit and we can instantiate the lease manager in main along with the rest of the process' lifecycle? Bonus: we don't need to do any logging or error handling.

Reply:
Yep this is a great call. I've moved the resource to the install manifests in https://github.com/linkerd/linkerd2/pull/10424/commits/a0a00582cba5fc1bdfc818b8befc3327097e1eb1 which removed the need for that function.